subsys/net/l2/openthread/Kconfig

```diff
@@ -2,110 +2,23 @@
 
 # Copyright (c) 2018 Nordic Semiconductor ASA
 # SPDX-License-Identifier: Apache-2.0
+
 menuconfig NET_L2_OPENTHREAD
 	bool "OpenThread L2"
 	depends on NETWORKING
-
-choice NET_L2_OPENTHREAD_IMPLEMENTATION
-	prompt "OpenThread L2 implementation"
-	depends on NET_L2_OPENTHREAD
-
-config NET_L2_OPENTHREAD_IMPLEMENTATION_ZEPHYR
-	bool "Zephyr's OpenThread L2 implementation"
-	select NET_L2_PHY_IEEE802154
 	select OPENTHREAD
 
-	imply FLASH
-	imply FLASH_MAP
-	imply MPU_ALLOW_FLASH_WRITE
-
-	select SETTINGS if FLASH
-	select OPENTHREAD_SETTINGS_RAM if !FLASH
-	select CPP
-	select REBOOT
-	select ENTROPY_GENERATOR
-
+	imply NET_L2_PHY_IEEE802154
```

Comment:
Why do we want to change select to imply for `NET_L2_PHY_IEEE802154`?

Reply:
I thought we would like to have this config editable if someone has a custom IEEE802.15.4 PHY, but I re-checked it, and another config, not this one, sets the PHY driver. I'll leave `select` here.